File: app/api/wizard.py
import json
from fastapi import APIRouter, Form
from app.db import get_db, mark_setup_complete

router = APIRouter()

@router.post("/submit")
async def submit_wizard(
    # Proxmox
    proxmox_endpoint: str = Form(""),
    proxmox_token_id: str = Form(""),
    proxmox_token_secret: str = Form(""),
    track_vms: int = Form(1),
    track_temps: int = Form(1),
    # Docker
    docker_hosts_json: str = Form("[]"),
    # Services
    use_grafana: int = Form(0),
    grafana_url: str = Form(""),
    grafana_api_key: str = Form(""),
    use_immich: int = Form(0),
    immich_url: str = Form(""),
    immich_api_key: str = Form(""),
    use_nut: int = Form(0),
    nut_url: str = Form(""),
    nut_device: str = Form("ups"),
    nut_username: str = Form("monuser"),
    nut_password: str = Form("")
):
    try:
        docker_hosts = json.loads(docker_hosts_json)
    except json.JSONDecodeError:
        docker_hosts = []

    async with get_db() as db:
        # Proxmox
        if proxmox_endpoint and proxmox_token_id and proxmox_token_secret:
            config = json.dumps({
                "track_vms": bool(track_vms),
                "track_temps": bool(track_temps)
            })
            secrets = json.dumps({
                "token_id": proxmox_token_id,
                "token_secret": proxmox_token_secret
            })
            await db.execute(
                "INSERT INTO data_sources (type, name, endpoint, config, secret_json) VALUES (?, ?, ?, ?, ?)",
                ("proxmox", "Proxmox Cluster", proxmox_endpoint.rstrip("/"), config, secrets)
            )

        # Docker hosts
        for host in docker_hosts:
            name = host.get("name", "Docker Host")
            host_ip = host.get("host", "")
            port = host.get("port", 2376)
            use_tls = host.get("use_tls", True)
            ca_cert = host.get("ca_cert", "")
            if host_ip:
                config = json.dumps({
                    "name": name,
                    "host": host_ip,
                    "port": int(port),
                    "use_tls": bool(use_tls),
                    "ca_cert": ca_cert
                })
                await db.execute(
                    "INSERT INTO data_sources (type, name, endpoint, config, secret_json) VALUES (?, ?, ?, ?, ?)",
                    ("docker", name, f"https://{host_ip}:{port}" if use_tls else f"http://{host_ip}:{port}", config, "{}")
                )

        # Grafana
        if use_grafana and grafana_url and grafana_api_key:
            config = json.dumps({"url": grafana_url.rstrip("/")})
            secrets = json.dumps({"api_key": grafana_api_key})
            await db.execute(
                "INSERT INTO data_sources (type, name, endpoint, config, secret_json) VALUES (?, ?, ?, ?, ?)",
                ("grafana", "Grafana", grafana_url.rstrip("/"), config, secrets)
            )

        # Immich
        if use_immich and immich_url and immich_api_key:
            config = json.dumps({"url": immich_url.rstrip("/")})
            secrets = json.dumps({"api_key": immich_api_key})
            await db.execute(
                "INSERT INTO data_sources (type, name, endpoint, config, secret_json) VALUES (?, ?, ?, ?, ?)",
                ("immich", "Immich", immich_url.rstrip("/"), config, secrets)
            )

        # NUT
        if use_nut and nut_url:
            config = json.dumps({"url": nut_url.rstrip("/"), "device": nut_device})
            secrets = json.dumps({"username": nut_username, "password": nut_password})
            await db.execute(
                "INSERT INTO data_sources (type, name, endpoint, config, secret_json) VALUES (?, ?, ?, ?, ?)",
                ("nut", "UPS (NUT)", nut_url.rstrip("/"), config, secrets)
            )

        await db.commit()

    await mark_setup_complete()
    return {"ok": True}

File: app/static/js/wizard.js
/* Wizard JavaScript */

let currentStep = 1;
let dockerHostCount = 1;

function goToStep(n) {
    currentStep = n;
    document.querySelectorAll(".step").forEach(s => {
        const sn = parseInt(s.dataset.step);
        s.classList.remove("active", "done");
        if (sn < n) s.classList.add("done");
        if (sn === n) s.classList.add("active");
    });
    document.querySelectorAll(".step-panel").forEach(p => {
        p.classList.toggle("active", parseInt(p.dataset.step) === n);
    });
}

function toggleFields(name) {
    const checkbox = document.getElementById("use-" + name);
    const fields = document.getElementById(name + "-fields");
    fields.style.display = checkbox.checked ? "block" : "none";
}

function addDockerHost() {
    dockerHostCount++;
    const container = document.getElementById("docker-hosts");
    const entry = document.createElement("div");
    entry.className = "docker-host-entry";
    entry.innerHTML = `
        <div style="font-size:13px;font-weight:500;margin-bottom:8px;">Host ${dockerHostCount}</div>
        <div class="form-group">
            <label class="form-label">Friendly Name</label>
            <input class="form-input" type="text" placeholder="e.g. Media Server" />
        </div>
        <div class="form-row">
            <div class="form-group">
                <label class="form-label">Host / IP</label>
                <input class="form-input" type="text" placeholder="10.10.x.x" />
            </div>
            <div class="form-group" style="flex:0 0 80px;">
                <label class="form-label">Port</label>
                <input class="form-input" type="text" value="2376" />
            </div>
        </div>
        <div class="checkbox-group">
            <input type="checkbox" checked />
            <label>Use TLS</label>
        </div>
    `;
    container.appendChild(entry);
}

function generateSummary() {
    const container = document.getElementById("summary-content");
    let html = "";

    // Proxmox
    const endpoint = document.getElementById("proxmox-endpoint").value;
    const tokenId = document.getElementById("proxmox-token-id").value;
    if (endpoint && tokenId) {
        html += `<div class="summary-section">
            <div class="summary-title">Proxmox Cluster</div>
            <div class="summary-row"><strong>Endpoint:</strong> ${escapeHtml(endpoint)}</div>
            <div class="summary-row"><strong>Token:</strong> ${escapeHtml(tokenId)}</div>
        </div>`;
    }

    // Docker
    const hosts = collectDockerHosts();
    if (hosts.length > 0) {
        html += `<div class="summary-section"><div class="summary-title">Docker Hosts</div>`;
        hosts.forEach(h => {
            html += `<div class="summary-row"><strong>${escapeHtml(h.name || 'Host')}</strong>: ${escapeHtml(h.host)}:${h.port}</div>`;
        });
        html += `</div>`;
    }

    // Services
    const services = [];
    if (document.getElementById("use-grafana").checked) services.push("Grafana");
    if (document.getElementById("use-immich").checked) services.push("Immich");
    if (document.getElementById("use-nut").checked) services.push("NUT / UPS");

    if (services.length > 0) {
        html += `<div class="summary-section">
            <div class="summary-title">Services</div>
            <div class="summary-row"><strong>Enabled:</strong> ${services.join(", ")}</div>
        </div>`;
    }

    container.innerHTML = html || "<p class='empty-state'>No data sources configured. You'll be able to add them in the Admin panel later.</p>";
    goToStep(4);
}

function collectDockerHosts() {
    const entries = document.querySelectorAll("#docker-hosts .docker-host-entry");
    const hosts = [];
    entries.forEach(entry => {
        const inputs = entry.querySelectorAll("input");
        const name = inputs[0].value.trim();
        const host = inputs[1].value.trim();
        const port = inputs[2].value.trim() || "2376";
        const useTls = inputs[3].checked;
        if (host) {
            hosts.push({ name, host, port, use_tls: useTls });
        }
    });
    return hosts;
}

async function submitWizard() {
    const btn = event.target;
    btn.disabled = true;
    btn.textContent = "Saving...";

    try {
        const fd = new FormData();
        // Proxmox
        fd.append("proxmox_endpoint", document.getElementById("proxmox-endpoint").value);
        fd.append("proxmox_token_id", document.getElementById("proxmox-token-id").value);
        fd.append("proxmox_token_secret", document.getElementById("proxmox-token-secret").value);
        fd.append("track_vms", document.getElementById("track-vms").checked ? 1 : 0);
        fd.append("track_temps", document.getElementById("track-temps").checked ? 1 : 0);
        // Docker
        fd.append("docker_hosts_json", JSON.stringify(collectDockerHosts()));
        // Services
        fd.append("use_grafana", document.getElementById("use-grafana").checked ? 1 : 0);
        fd.append("grafana_url", document.getElementById("grafana-url").value);
        fd.append("grafana_api_key", document.getElementById("grafana-api-key").value);
        fd.append("use_immich", document.getElementById("use-immich").checked ? 1 : 0);
        fd.append("immich_url", document.getElementById("immich-url").value);
        fd.append("immich_api_key", document.getElementById("immich-api-key").value);
        fd.append("use_nut", document.getElementById("use-nut").checked ? 1 : 0);
        fd.append("nut_url", document.getElementById("nut-url").value);
        fd.append("nut_device", document.getElementById("nut-device").value);
        fd.append("nut_username", document.getElementById("nut-username").value);
        fd.append("nut_password", document.getElementById("nut-password").value);

        const res = await fetch("/api/wizard/submit", {
            method: "POST",
            body: fd
        });

        if (res.ok) {
            window.location.href = "/";
        } else {
            const err = await res.text();
            alert("Error saving configuration: " + err);
            btn.disabled = false;
            btn.textContent = "Save & Start Dashboard";
        }
    } catch (e) {
        alert("Error: " + e.message);
        btn.disabled = false;
        btn.textContent = "Save & Start Dashboard";
    }
}

function escapeHtml(str) {
    const div = document.createElement("div");
    div.textContent = str || "";
    return div.innerHTML;
}
